### ISSUE TITLE
Alt modifier not sent for some symbols

### ISSUE THREAD

**Issue by occivink** [kovidgoyal/kitty#675]
using archlinux, kitty 0.11.1, x11 and default config

When pressing certain `alt + KEY` combinations, the program inside kitty does not receive the alt modifier. This is reproducible when KEY is any of `'`  `"`  `<`  `>`  `.`  ` ,` `/` `?`.

In addition, after pressing any of these combinations, the alt modifier is not sent anymore until the alt key is raised and depressed again. For example:
alt+a -> program receives `^[a`
alt+< -> program receives `<`
alt+a -> program receives `a`

I used `showkey -a` to identify the keys sent.

**Comment by kovidgoyal** [comment]
I cannot reproduce on my arch system. Pressing alt+a three times produces the follwing output in showkey -a:

```
Press any keys - Ctrl-D will terminate this program

^[a      27 0033 0x1b
         97 0141 0x61
^[a      27 0033 0x1b
         97 0141 0x61
^[a      27 0033 0x1b
         97 0141 0x61
```

This will be some "feature" of your keyboard layout. Switch to a default keyboard layout and you should be fine. Additionally, you can see exactly what key events kitty receives and how it interprets them witht he ``--debug-keyboard`` command line option.

**Comment by occivink** [comment]
I'm not using any special keyboard layout, only en_US. 
However, I think there is a misunderstanding in the issue, this is about alt+symbol with certain non-alphanumeric symbols. 

As you can see in this `--debug-keyboard` snippet, the modifier is not sent when pressing `.` despite alt being down.
```
scancode: 0x40 release: 0 clean_sym: Alt_L composed_sym: Alt_L mods: none glfw_key: LEFT ALT
scancode: 0x3c release: 0 clean_sym: period composed_sym: period text: . mods: none glfw_key: PERIOD
scancode: 0x3c release: 1 clean_sym: period mods: none glfw_key: PERIOD
scancode: 0x40 release: 1 clean_sym: Alt_L mods: none glfw_key: LEFT ALT
```

**Comment by kovidgoyal** [comment]
Cannot reproduce that either:
```
scancode: 0x40 release: 0 clean_sym: Alt_L composed_sym: Alt_L mods: none glfw_key: LEFT ALT xkb_key_name: Alt_L
scancode: 0x3c release: 0 clean_sym: period composed_sym: period mods: alt glfw_key: PERIOD xkb_key_name: period
scancode: 0x3c release: 1 clean_sym: period mods: alt glfw_key: PERIOD xkb_key_name: period
scancode: 0x3c release: 0 clean_sym: period composed_sym: period mods: alt glfw_key: PERIOD xkb_key_name: period
scancode: 0x3c release: 1 clean_sym: period mods: alt glfw_key: PERIOD xkb_key_name: period
scancode: 0x3c release: 0 clean_sym: period composed_sym: period mods: alt glfw_key: PERIOD xkb_key_name: period
scancode: 0x3c release: 1 clean_sym: period mods: alt glfw_key: PERIOD xkb_key_name: period
scancode: 0x3c release: 0 clean_sym: period composed_sym: period mods: alt glfw_key: PERIOD xkb_key_name: period
scancode: 0x3c release: 1 clean_sym: period mods: alt glfw_key: PERIOD xkb_key_name: period
```

**Comment by occivink** [comment]
For what it's worth, I cannot reproduce any of this on my laptop that has an almost identical setup. However, the computer on which I'm experiencing this issue can handle these key combinations just fine in other terminal emulators, so I really have no idea what's causing this

**Comment by kovidgoyal** [comment]
Well if you want to debug in more detail, the relevant code is all in the glfw_xkb_handle_key_event() function. IMO, this is likely caused by a funky xkb layout.

**Comment by occivink** [comment]
For what it's worth, I discovered that this was caused by my keyboard. It has an n-key rollover feature, which causes the keyboard to be reported as multiple input devices. In particular, the modifier keys belong to one input device, and the other keys I mentioned (`'` `"` `<` `>` `.` `,` `/` `?`...) belong to an other input device. It looks like modifiers are not sent if they belong to a different device.

I guess glfw's input handling is to blame in this case.
